dll/shellext/cabview/folder.cpp

```diff
@@ -0,0 +1,859 @@
+/*
+ * PROJECT:     ReactOS CabView Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Shell folder implementation
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "cabview.h"
+#include "util.h"
+
+enum FOLDERCOLUMNS
+{
+    COL_NAME,   // PKEY_ItemNameDisplay
+    COL_SIZE,   // PKEY_Size
+    COL_TYPE,   // PKEY_ItemTypeText
+    COL_MDATE,  // PKEY_DateModified
+    COL_PATH,   // PKEY_?: Archive-relative path
+    COL_ATT,    // PKEY_FileAttributes
+    COLCOUNT
+};
+
+static const struct FOLDERCOLUMN
+{
+    BYTE TextId;
+    BYTE LvcFmt;
+    BYTE LvcChars;
+    BYTE ColFlags;
+    const GUID *pkg;
+    BYTE pki;
+} g_Columns[] =
+{
+    { IDS_COL_NAME, LVCFMT_LEFT, 20, SHCOLSTATE_TYPE_STR | SHCOLSTATE_ONBYDEFAULT, &FMTID_Storage, PID_STG_NAME },
+    { IDS_COL_SIZE, LVCFMT_RIGHT, 16, SHCOLSTATE_TYPE_INT | SHCOLSTATE_ONBYDEFAULT, &FMTID_Storage, PID_STG_SIZE },
+    { IDS_COL_TYPE, LVCFMT_LEFT, 20, SHCOLSTATE_TYPE_STR | SHCOLSTATE_ONBYDEFAULT, &FMTID_Storage, PID_STG_STORAGETYPE },
+    { IDS_COL_MDATE, LVCFMT_LEFT, 20, SHCOLSTATE_TYPE_DATE | SHCOLSTATE_ONBYDEFAULT, &FMTID_Storage, PID_STG_WRITETIME },
+    { IDS_COL_PATH, LVCFMT_LEFT, 30, SHCOLSTATE_TYPE_STR | SHCOLSTATE_ONBYDEFAULT, &CLSID_CabFolder, 0 },
+    { IDS_COL_ATT, LVCFMT_RIGHT, 10, SHCOLSTATE_TYPE_STR, &FMTID_Storage, PID_STG_ATTRIBUTES },
+};
+
+#include <pshpack1.h>
+struct CABITEM
+{
+    WORD cb;
+    WORD Unknown; // Not sure what Windows uses this for, we always store 0
+    UINT Size;
+    WORD Date, Time; // DOS
```

Comment:
In which order are they stored? First Date, then Time? Or the other way around?

Reply:
Date first yes. Same concept as FileStruct in shell32.